tutor/hooks/__init__.py

```diff
@@ -3,14 +3,16 @@
 
 import typing as t
 
+from . import actions as _actions
+from . import filters as _filters
```

Comment:
Ah, yes. You mean that this does not resolve the confusion around `hooks.Actions` vs `hooks.actions`, right? In your [other comment](https://github.com/openedx/wg-developer-experience/issues/125#issuecomment-1369973269) you mentioned that we should flatten the actions.py/filters.py modules into `__init__.py`. But this would again cause confusion between `hooks.Actions` and `hooks.Action`, right?

My original sin here was that I wanted to expose the hooks catalog in `tutor.hooks`. I think that we should have gone with:

    import tutor.hooks.catalog as hooks

If you agree, then we can make this change in the plugin cookiecutter after this PR is merged. Then in Palm we can get rid of the hooks/__init__.py module altogether.

Reply:
Hm... stepping back, we're trying to balance three things here: clarity of the API, brevity of the import paths, and the backwards-compatibility of this this change. Let's look at the options we have for the most important public API elements.

(1) Currently, we have:
```
tutor.hooks.Actions
tutor.hooks.Filters
tutor.hooks.priorities.Priorities
tutor.hooks.actions.Action
tutor.hooks.filters.Filter
tutor.hooks.contexts.Context
```

(2) My comment suggested:
```
tutor.hooks.Actions
tutor.hooks.Filters
tutor.hooks.Priorities
tutor.hooks.Action
tutor.hooks.Filter
tutor.hooks.Context
```

(3) Your comment suggests:
```
tutor.hooks.catalog.Actions
tutor.hooks.catalog.Filters
tutor.hooks.priorities.Priorities
tutor.hooks.actions.Action
tutor.hooks.filters.Filter
tutor.hooks.contexts.Context
```

I think (2) wins on brevity, but (3) is clearer in that it separates the "catalog" classes (Actions/Filters) out from the "type" classes (Action/Filter/Context). In either case, we changing a handful of import paths, and could maintain backwards-compatibility until Palm.

I'll throw one last suggestion in: (4)
```
tutor.hooks.Actions
tutor.hooks.Filters
tutor.hooks.Priorities
tutor.hooks.types.Actions
tutor.hooks.types.Filter
tutor.hooks.types.Context
```

I like this one because it separates the catalog classes from the type classes, but the imports for the catalog classes (including Priorities) are shorter. As for backwards-compatibility, we are changing 3 import paths, but it would be easy to maintain compatibility until Palm: we'd have wrapper modules (actions.py, filters.py, contexts.py, priorities.py), each of which could raise a warning a la _Importing from ... is deprecated, please import ... instead_ and then re-export their old items from their canonical locations.

Let me know what you think. I'd like to make this decision carefully--this is our one shot to restructure the V1 API!  😄

Reply:
Your suggestion is better, but I don't like "types". I suggest that we start creating a new "core" module:

```
tutor.hooks.Actions
tutor.hooks.Filters
tutor.hooks.Priorities
tutor.core.hooks.Action
tutor.core.hooks.Filter
tutor.core.hooks.Context
```

After all, hook classes are very much at the "core" of Tutor now. Just looking at the imports we can see that they don't pull anything from outside the hooks framework, and are thus completely unrelated to Open edX.

Reply:
Are you thinking that tutor.core would be the eventual destination for other "tutorcore" things too, like plugins/ and config.py?

Reply:
Yes, exactly.

Reply:
Sounds good to me then 👍

Priorities is funny--I could see it go in either place. It feels like a part of the "catalog", but also, there's nothing openedx-specific about the priority constants.